libs/utils/android/benchmark.py

```diff
@@ -203,9 +211,15 @@ def __init__(self):
             raise
 
         self._log.info('=== Execution...')
-        self.wl.run(out_dir=self.out_dir,
-                    collect=self._getBmCollect(),
-                    **self.bm_params)
+        for iter in range(self.bm_iterations):
```

Comment:
This gets complicated by the fact that most often you need a first reboot (to reboot with your kernel) and you can decide after if you need subsequent reboots (between your different tests or iterations of your tests).

I way I think this can work is:
 - The first boot is conditioned by the presence of the boot image as a command line argument.
    That will happen even if a "--reboot" is not passed (see below for meaning of this new argument)
 - Reboots between tests or iterations is controlled by an additional "--reboot" command line argument.
 - A possible delay  is conditioned by "--boot-timeout" (maybe we should change the string) even if "--reboot" is present. I'm wondering here if it's best to add a default value(60s) to this configured argument if --reboot is present. Because if people forget to pass this when they require reboot then we don't wait for the reboot to complete and the call to the benchmark will fail.

This will mean that we remove all handling of a reboot in the test scripts together with the delay after each test run, and we add the following to benchmark.py:
 - reboot_target in __init__ if a boot image is given.
 - a call of this function (maybe called prepare_run) that would include (default):
     - - call to reboot_target if --reboot is present
     - - wait for a period of time if present (boot_timeout)

Both --reboot and --boot_timeout can be configured from Jenkins.